src/alcotest-engine/test.ml

```diff
@@ -177,6 +177,41 @@ let pp_location =
     | None, None -> ()
 
 let check (type a) ?here ?pos (t : a testable) msg (expected : a) (actual : a) =
+  let here =
+    match here with
+    | Some here -> Some here
+    | None ->
+        (* Best-effort reporting of the location of the call to Alcotest.check. *)
+        let open Printexc in
+        let callstack_depth = 3 in
+        let raw_backtrace = get_callstack callstack_depth in
+        let len = raw_backtrace_length raw_backtrace in
+        if len >= 3 then
+          let slot0 =
+            get_raw_backtrace_slot raw_backtrace 0 |> convert_raw_backtrace_slot
+          and slot2 =
+            get_raw_backtrace_slot raw_backtrace 2 |> convert_raw_backtrace_slot
+          in
+          match (Slot.name slot0, Slot.name slot2) with
+          | ( Some "Alcotest_engine__Test.check",
+              Some "Alcotest_engine__Core.Make.protect_test.(fun)" ) -> (
```

Comment:
One problem I see here might be that there can be multiple call layers between `Core.Make.protect_test` and `Test.check`. 
For example if the user's test calls into an intermediate function that calls into `check`, then `protect_test` would be present in slot `3` instead of `2`.

Maybe it's better do that the other way around: check that slot `0` is `check` and that `1` is _not_ `protect_test` ?

Reply:
You're right! I've changed that.